mod/cornercutter/ModFeature/ModSetting/MultiSpawnerSwapper.cs

```diff
@@ -0,0 +1,57 @@
+﻿using cornercutter.Enum;
+using cornercutter.ModLoader;
+using HarmonyLib;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace cornercutter.ModFeature.ModSetting
+{
+    [HarmonyPatch(typeof(MultiSpawner), nameof(MultiSpawner.Spawn))]
+    class MultiSpawnerSwapper
+    {
+        static void Prefix(ref MultiSpawner __instance)
+        {
+            if (!CutterConfig.Instance.CornercutterIsEnabled()) return;
+
+            MultiSpawnerType multiSpawnerType = CutterConfig.Instance.MultiSpawnerType;
+            List<EntitySpawner> spawners = __instance.subSpawners;
+            int[] spawnerWeights = __instance.subSpawnerWeights;
+
+            int totalWeight = 0;
+            int[] newWeights = new int[spawnerWeights.Length];
+
+            for (int i = 0; i < spawners.Count; i++)
+            {
+                EntitySpawner spawner = spawners[i];
+                if (spawner)
+                {
+                    int originalWeight = spawnerWeights[i];
+                    int newWeight;
+
+                    switch (multiSpawnerType)
+                    {
+                        case MultiSpawnerType.AlwaysSkillIfAble:
+                            newWeight = spawner.name == "SkillSpawn" ? originalWeight : 0;
+                            break;
+                        case MultiSpawnerType.Never:
+                            newWeight = spawner.name == "SkillSpawn" ? 0 : originalWeight;
+                            break;
+                        default:
+                            newWeight = originalWeight;
+                            break;
+                    }

```

Comment:
This looks good to me, but if you did want to condense it, there are only 2 ways I can think of:
1. Having a `MultiSpawner` interface and subclasses which could implement each different case. Probably quite a bulky solution though.
2. A condensed switch:
```cs
int newWeight = multiSpawnerType switch {
   MultiSpawnerType.AlwaysSkillIfAble => spawner.name == "SkillSpawn" ? originalWeight : 0,
   MultiSpawnerType.Never             => spawner.name == "SkillSpawn" ? 0 : originalWeight,
   _                                  => originalWeight,
};
```
Reference: https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression

It's probably not necessary though and considering this is how you've been doing switch statements everywhere else it should be fine.

Reply:
This is much cleaner but is unfortunately a C# 8.0 feature, and this project is built against .NET Standard 2.0 (C# 7.3). To avoid any upgrade hassles this can be flagged for cleanup for the time being and revisited.